ddtrace/internal/forksafe.py

```diff
@@ -3,39 +3,45 @@
 """
 import logging
 import os
-
-
-__all__ = [
-    "ddtrace_after_in_child",
-    "register",
-]
+import typing
 
 
 log = logging.getLogger(__name__)
 
 
-registry = []
+_registry = []
 
 
 def ddtrace_after_in_child():
-    global registry
+    # type: () -> None
+    global _registry
 
-    for hook in registry:
+    for hook in _registry:
         try:
             hook()
         except Exception:
             # Mimic the behaviour of Python's fork hooks.
             log.exception("Exception ignored in forksafe hook %r", hook)
 
 
-def _register(hook):
-    if hook not in registry:
-        registry.append(hook)
+def register(
+    after_in_child,  # type: typing.Callable
+):
+    """Register a function to be called after fork in the child process."""
+    # type: (...) -> None
```

Comment:
Is this typing annotation for the `register()` function? If so, should it be returning a `Callable`?

Reply:
should probably go above the docstring too

Reply:
fixed